src/main/kotlin/com/terraformation/backend/accelerator/api/GlobalRolesController.kt

```diff
@@ -0,0 +1,75 @@
+package com.terraformation.backend.accelerator.api
+
+import com.terraformation.backend.api.AcceleratorEndpoint
+import com.terraformation.backend.api.ApiResponse200
+import com.terraformation.backend.api.ApiResponse404
+import com.terraformation.backend.api.SimpleSuccessResponsePayload
+import com.terraformation.backend.api.SuccessResponsePayload
+import com.terraformation.backend.customer.db.UserStore
+import com.terraformation.backend.customer.model.IndividualUser
+import com.terraformation.backend.customer.model.requirePermissions
+import com.terraformation.backend.db.default_schema.GlobalRole
+import com.terraformation.backend.db.default_schema.UserId
+import io.swagger.v3.oas.annotations.Operation
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RestController
+
+@AcceleratorEndpoint
+@RequestMapping("/api/v1/accelerator/globalRoles")
+@RestController
+class GlobalRolesController(
+    private val userStore: UserStore,
+) {
+  @ApiResponse200
+  @GetMapping("/users")
+  @Operation(summary = "Gets the list of users that have global roles.")
+  fun listGlobalRoles(): GlobalRoleUsersListResponsePayload {
+    requirePermissions { readGlobalRoles() }
+
+    return GlobalRoleUsersListResponsePayload(
+        userStore.fetchWithGlobalRoles().map { UserWithGlobalRolesPayload(it) })
+  }
+
+  @ApiResponse200
+  @ApiResponse404
+  @PostMapping("/users/{userId}")
+  @Operation(summary = "Apply the supplied global roles to the user.")
+  fun updateGlobalRoles(
+      @PathVariable("userId") userId: UserId,
+      @RequestBody payload: UpdateGlobalRolesRequestPayload,
+  ): SuccessResponsePayload {
+    val roleEnums = payload.globalRoles.map { GlobalRole.forJsonValue(it) }.toSet()
+
+    userStore.updateGlobalRoles(userId, roleEnums)
+
+    return SimpleSuccessResponsePayload()
+  }
+}
+
+data class UserWithGlobalRolesPayload(
+    val id: UserId,
+    val email: String,
+    val firstName: String?,
+    val globalRoles: Set<GlobalRole>,
+    val lastName: String?,
+) {
+  constructor(
+      user: IndividualUser
+  ) : this(
+      id = user.userId,
+      email = user.email,
+      firstName = user.firstName,
+      globalRoles = user.globalRoles,
+      lastName = user.lastName)
+}
+
+data class GlobalRoleUsersListResponsePayload(val users: List<UserWithGlobalRolesPayload>) :
+    SuccessResponsePayload
+
+data class UpdateGlobalRolesRequestPayload(
+    val globalRoles: List<String>,
```

Comment:
Did you try making this `Set<GlobalRole>` instead? I believe that should work, but maybe you tried it and it didn't. That way you don't need to do the string-to-enum conversion in application code, and as an added bonus, the OpenAPI schema should include a list of allowed values.

Reply:
I changed it to `Set<GlobalRole>`